packages/syntax-core/src/Modal/Modal.tsx

```diff
@@ -236,7 +243,11 @@ export default function Modal({
                 >
                   {header}
                 </Heading>
-                <Box marginBottom={themeName === "classic" ? 4 : 0}>
+                <Box
+                  marginBottom={themeName === "classic" ? 4 : 0}
+                  maxHeight={"calc(100vh - 232px)"} // 232x is combined height of header/footer/padding/margin
```

Comment:
@christianvuerings i know you're not a fan of these "magic" numbers. but `maxHeight` is required to get this scrollbar to show up properly, and i think this is "exactly" what is we want here.

I don't know a way of doing a dynamic max height other than like this?

Reply:
but for documentation's sake:
the math is:

48px header height
16px gap between body/header
48px footer
8px gap between body/footer
32 top margin
32 bottom margin
24 top padding
24 bottom padding

Reply:
would % work for it? I used it in `LessonV2DetailsModal` but not sure that would work in this context.

```
      <Box
        maxHeight="80%"
        overflowY="auto"
        dangerouslySetInlineStyle={{
          __style: { scrollbarWidth: "none", paddingBottom: "36px" },
        }}
        gap={3}
        direction="column"
        display="flex"
        marginTop={3}
      >
```

Reply:
@somethiiing yes, we should be able to do this without this magic number. Main thing is that it's really hard to keep this number updated if any of the components inside of it change.

Please take a look at https://github.com/Cambly/syntax/pull/367 for an alternative approach